elastic/kibana
# [ResponseOps] only allow HTML email via notifications

resolves https://github.com/elastic/response-ops-team/issues/122

The email connector is changed to only allow HTML email to be sent via notifications - not via alerting rule actions or via the http execute route.

## To test

To test, do the following set up:

- run `npx maildev` to start a local mail server

- add the following to your `kibana.dev.yaml`:

      notifications:
        connectors:
          default:
            email: maildev-no-auth
      #
      xpack.actions.preconfigured:
        maildev-no-auth:
          name: 'email: maildev no auth'
          actionTypeId: '.email'
          config:
            from: 'me@example.com'
            host: 'localhost'
            port: '1025'


- create a new role that includes the Kibana Security/Cases feature

- create a new user that has that role

- login to that user to get profile created - easiest done in an incognito window so you don't have to log out of `elastic`

- back in your `elastic` user session, create a case

- assign it to the new user

- check your email, it should be nice and pretty

- now run the following curl command:

      curl -X POST $KB_URL/api/actions/connector/maildev-no-auth/_execute \
          -H 'kbn-xsrf: true' \
          -H 'Content-Type: application/json' \
          -d '
            { 
              "params": {
                "to": ["me@example.com"],
                "subject": "testing html email",
                "message": "plain text",
                "messageHTML": "html text"
              }
            }
          '
- you should see the following output:

      {"status":"error","message":"HTML email can only be sent via notifications","connector_id":"maildev-no-auth"}
